src/components/NotFoundPage/NotFoundPage.tsx

```diff
@@ -7,58 +7,39 @@ import Footer from 'components/Footer'
 import Navbar from 'components/Navbar'
 import LoadingPage from 'components/LoadingPage'
 
-import { Props, State } from './NotFoundPage.types'
+import { Props } from './NotFoundPage.types'
 import './NotFoundPage.css'
 
-const LOADING_TIMEOUT = 2000
+export default function NotFoundPage(props: Props) {
+  const [isLoading, setIsLoading] = useState<Boolean>(true);
 
-export default class NotFoundPage extends React.PureComponent<Props, State> {
-  state = {
-    isLoading: true
-  }
-
-  mounted = false
+  useEffect(() => {
+    const analytics = getAnalytics()
+    analytics.track('Not found page', {})
 
-  analytics = getAnalytics()
+    setIsLoading(false);
+  }, [])
 
-  componentWillMount() {
-    document.body.classList.add('notfound-body')
-    this.analytics.track('Not found page', {})
-    this.mounted = true
-    this.setState({ isLoading: true })
-    setTimeout(() => {
-      if (this.mounted) {
-        this.setState({ isLoading: false })
-      }
-    }, LOADING_TIMEOUT)
```

Comment:
@cazala Do you think this logic'd be necessary on this new state-less version? it seems like it's trying to **for sure** see if the component was rendered

Reply:
mm i _think_ the idea for this timeout was to avoid a state that happened when you refreshed the editor and you would see a "not found" page for a bit, until it started loading the project. It's not a big deal and the solution was not pretty so it's not a big concern on my side if we removed it.